celery_utils/logged_task.py

```diff
@@ -37,5 +37,11 @@ def on_retry(self, exc, task_id, args, kwargs, einfo):
         Capture the exception that caused the task to be retried, if any.
         """
         super(LoggedTask, self).on_retry(exc, task_id, args, kwargs, einfo)
-        if exc is not None:
-            log.warning('[{}] retried due to {}'.format(task_id, einfo.traceback))
+        log.warning('[{}] retried due to {}'.format(task_id, getattr(einfo, 'traceback', None)))
```

Comment:
Are there cases where retries happen due to reasons other than an exception being thrown? (otherwise, we could be spamming the logs with a lot of "[...] retried due to None")

Reply:
Good question.  It didn't look that way to me: http://docs.celeryproject.org/en/latest/userguide/tasks.html#retrying

Reply:
I'll keep an eye out on Splunk - just to be sure though.